### ISSUE TITLE
Error in hclust(d = d.dist, method = "average") :    NA/NaN/Inf in foreign function call (arg 10)

### ISSUE THREAD

**Issue by Tswunami** [lmc297/bactaxR#4]
I have successfully ran the package to generate a histogram, however when I am trying to generate the dendrogram I ran into the following error: 

> summary(ani)
An object of class bactaxRObject 
Number of genomes:  60 
Total number of pairwise ANI comparisons:  2231
> h
> dend <- ANI.dendrogram(bactaxRObject = ani, ANI_threshold = 94.2, xline = c(4,5,6,7.5), xlinecol = c("#ffc425", "#f37735", "deeppink4", "black"), label_size = 0.5)
Error in hclust(d = d.dist, method = "average") : 
  NA/NaN/Inf in foreign function call (arg 10)

I have checked my ani input does not contain any NA/NaN/Inf. 

Please kindly help, thanks in advance.

**Comment by lmc297** [comment]
Hi Tswunami! Thanks so much for using bactaxR (and sorry for not responding sooner!)
I think your issue might be related to the fact that there are not pairwise comparisons for all genomes; for example, if you have 60 genomes, I would expect there to be 60*60 = 3600 ANI values. It looks like there are only 2231 in your data set (I'll try to edit bactaxR to catch this in the future and throw an error)

**Comment by Tswunami** [comment]
Thank you very much Laura! I have successfully generated the dendrogram using the test dataset provided. I have made sure the input has all pairwise comparison ANI values
However, when trying to run with my actual dataset, I encountered another issue with ANI.dendrogram. 

> summary (ani)
An object of class bactaxRObject 
Number of genomes:  8439 
Total number of pairwise ANI comparisons:  71233600
> h <- ANI.histogram(bactaxRObject = ani, bindwidth = 0.1)
> h
> dend <- ANI.dendrogram(bactaxRObject = ani, ANI_threshold = 95, xline = c(4,5,6,7.5), xlinecol = c("#ffc425", "#f37735", "deeppink4", "black"), label_size = 0.5)
Aggregation function missing: defaulting to length
Error: C stack usage  15927104 is too close to the limit

My dataset has a large number of genomes, can I change the limit of the ANI.dendrogram so I can run the package with a larger dataset?

Kind regards 

Tswunami
